core/src/main/java/hudson/Functions.java

```diff
@@ -574,7 +574,7 @@ public static String[] printLogRecordHtml(LogRecord r, LogRecord prior) {
     private static String[] logRecordPreformat(LogRecord r) {
         String source;
         if (r.getSourceClassName() == null) {
-            source = r.getLoggerName();
+            source = r.getLoggerName() == null ? "" : r.getLoggerName();
```

Comment:
tip: `Util.fixNull`

Reply:
@jglick thanks, noted for next time. By the way do you know why [these](https://github.com/jglick/jenkins/blob/d3dc69f8b7270e885f0c055b2210706fc5411c96/core/src/main/java/hudson/util/RingBufferLogHandler.java#L106) `<discarded>` log records show up in the log UI? This PR just makes sure they can be displayed without NPEs, but a proper fix would be to make sure they don't show up at all...

Reply:
The `<discarded>` log records were introduced in #6044, which was itself an amendment to #6018. See those two PRs for the reasoning behind this change.